nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -852,12 +1013,60 @@ impl DataStoreInventoryTest for DataStore {
                     .context("failed to allow table scan")?;
 
                 use db::schema::inv_collection::dsl;
-                dsl::inv_collection
+                let collection_ids = dsl::inv_collection
                     .select(dsl::id)
                     .order_by(dsl::time_started)
                     .load_async(&conn)
                     .await
-                    .context("failed to list collections")
+                    .context("failed to list collections")?;
+
+                // Ensure we don't have any stray collection IDs in other
+                // inventory tables that reference collection IDs that are not
+                // present in `inv_collection`.
```

Comment:
This is the first of the two test issues I mentioned in the PR description. I had forgotten to remove rows in `inv_root_of_trust_page` when deleting a collection, but the existing tests still passed. I added this bit to check that we don't have any leftover collection IDs in tables other than `inv_collection`, but maybe this should be a separate method that the tests call explicitly? Open to suggestions.

Reply:
Yeah I thought about this too but couldn't come up with a great way to solve the problem of forgetting to update the inventory delete code.  Would this test have caught your issue?  I wouldn't think so because I assume you'd also have forgotten to update the test to check your new table.

Reply:
It would not have caught my issue without updating the test to check the new table, no. I did make a pass through updating the tests in this file before realizing I'd forgotten to update the delete code, though, so maybe I would have added the table here and then seen a test failure? Hard to know, and I'm not sure whether keeping this around is worth it - thoughts?

I can only think of terrible ways to try to ensure we've checked all the tables that might hold dangling collection IDs, none of which are worth voicing aloud I don't think.

Reply:
I don't think it's worth keeping but I don't mind if you want to.

This is getting off-topic but the bestthing I've come to is this: this issue does come up a fair bit and I could imagine a mechanism where we take a database dump before and after a test and verify that it hasn't left detritus around.  This mechanism would have to be configurable since the system in general _does_ produce stuff unrelated to the test.  For example, every _other_ Nexus test will wind up having an inventory collection in the database after it runs and it should ignore that.  But these tests shouldn't.  Similarly, these tests should ignore stuff generated by other subsystems, but _not_ the inventory stuff.  I could imagine grouping tables into categories and having the mechanism say "this category should now be empty" or something.  Even this only really solves this problem if we enforce that every table _have_ a category so that we can't accidentally forget to add new tables like in the case here.

Anyway, I think this is probably a good idea and well beyond the scope of this PR.